javaServices/coreJavaServices/src/joliex/util/XmlUtils.java

```diff
@@ -64,35 +64,79 @@ public String valueToXml( Value request )
+			if ( request.hasChildren( "rootNodeName" ) ) {
+				rootNodeName = request.getFirstChild( "rootNodeName" ).strValue();
+			}
+				
+			boolean isXmlStore = true;
+			if ( request.hasChildren( "plain" ) ) {
+				isXmlStore = !request.getFirstChild( "plain" ).boolValue();
+			}
+			if ( request.hasChildren( "isXmlStore" ) ) {
+				isXmlStore = request.getFirstChild( "isXmlStore" ).boolValue();
+			}
+			
+			boolean indent = false;
+			if ( request.hasChildren( "indent" ) ) {
+				indent = request.getFirstChild( "indent" ).boolValue();
+			}
+			
+			boolean isApplySchema = false;
+			String schemaFilename = null;
+			String encoding = null;
+			String doctypeSystem = null;
+			if ( request.hasChildren( "applySchema" ) ) {
+				isApplySchema = true;
+				Value applySchema = request.getFirstChild( "applySchema" );
+				schemaFilename = applySchema.getFirstChild( "schema" ).strValue();
+				if ( applySchema.hasChildren( "encoding" ) ) {
+					encoding = applySchema.getFirstChild( "encoding" ).strValue();
+				}
+				if ( applySchema.hasChildren( "doctypeSystem" ) ) {
+					doctypeSystem = applySchema.getFirstChild( "doctypeSystem" ).strValue();
+				}
+			} 
+			
+			if ( !isXmlStore ) {
+				if ( isApplySchema ) {
+					transformer = jolie.xml.XmlUtils.valueToDocument( value, doc, schemaFilename, indent, doctypeSystem, encoding );
+				} else {
+					jolie.xml.XmlUtils.valueToDocument(
+						value,
+						rootNodeName,
+						doc
+					);
+					transformer = transformerFactory.newTransformer();
+					if ( indent ) { transformer.setOutputProperty( OutputKeys.INDENT, "yes" ); } 
+						else { transformer.setOutputProperty( OutputKeys.INDENT, "no" ); }
+				}
 			} else {
 				jolie.xml.XmlUtils.valueToStorageDocument(
-					request.getFirstChild( "root" ),
+					value,
 					rootNodeName,
 					doc
 				);
+				transformer = transformerFactory.newTransformer();
+				if ( indent ) { transformer.setOutputProperty( OutputKeys.INDENT, "yes" ); } 
+					else { transformer.setOutputProperty( OutputKeys.INDENT, "no" ); }
 			}
-			Transformer t = transformerFactory.newTransformer();
+		
+			
 			if ( request.getFirstChild( "omitXmlDeclaration" ).boolValue() ) {
-				t.setOutputProperty( OutputKeys.OMIT_XML_DECLARATION, "yes" );
+				transformer.setOutputProperty( OutputKeys.OMIT_XML_DECLARATION, "yes" );
 			}
-			if ( request.getFirstChild( "indent" ).boolValue() ) {
-				t.setOutputProperty( OutputKeys.INDENT, "yes" );
-			} else {
-				t.setOutputProperty( OutputKeys.INDENT, "no" );
-			}
-			StringWriter writer = new StringWriter();
-			StreamResult result = new StreamResult( writer );
-			t.transform( new DOMSource( doc ), result );
-			return writer.toString();
-		} catch( ParserConfigurationException e ) {
+			
+			StringWriter outWriter = new StringWriter();
+			StreamResult result = new StreamResult( outWriter );
+			transformer.transform( new DOMSource( doc ), result );
+			StringBuffer sb = outWriter.getBuffer();
+			return sb.toString();
```

Comment:
You can just invoke outWriter.toString()

Reply:
What about this? Could we merge the pull request?